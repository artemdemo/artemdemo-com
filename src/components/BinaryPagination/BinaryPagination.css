.pagination-container::after {
  content: '';
  display: table;
  clear: both;
}

@media (max-width: 768px) {
  .pagination-container {
    text-align: center;
  }
}

.pagination-item {
}

.pagination-item_left {
  float: left;
}

.pagination-item_right {
  float: right;
}

@media (max-width: 768px) {
  .pagination-item {
    float: initial;
    display: inline-block;
    width: 51%;
  }
}

@media (max-width: 768px) {
  .binary-pagination {
    display: block;
  }
}

.binary-pagination-link {
  max-width: 200px;
  display: flex;
}

@media (min-width: 992px) {
  .binary-pagination-link {
    max-width: 300px;
  }
}
