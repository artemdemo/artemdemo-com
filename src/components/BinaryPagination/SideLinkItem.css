.side-link-arrow {
  flex-grow: 1;
}
  .side-link-arrow:first-child {
    padding-right: 5px;
  }
  .side-link-arrow:last-child {
    padding-left: 5px;
  }

.side-link-text {
  text-overflow: ellipsis;
  white-space: nowrap;
  overflow: hidden;
}
