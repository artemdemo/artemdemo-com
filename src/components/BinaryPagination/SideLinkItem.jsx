import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import styled from 'styled-components';

export const SIDE_ITEM_DIRECTION = {
    'PREV': 'prev',
    'NEXT': 'next',
};

const LinkText = styled.span`
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
`;

const LinkArrow = styled.span`
    flex-grow: 1;
    :first-child {
        padding-right: 5px;
    }
    :last-child {
        padding-left: 5px;
    }
`;

const SideLinkItem = (props) => {
    const { direction, title, to } = props;
    const renderArrow = () => (
        <LinkArrow>
            {direction === SIDE_ITEM_DIRECTION.PREV ? '←' : '→'}
        </LinkArrow>
    );
    const renderTitle = () => (
        <LinkText>
            {title}
        </LinkText>
    );
    return (
        <ul className='pagination binary-pagination'>
            <li className='page-item'>
                <Link
                    className='page-link binary-pagination-link'
                    to={to}
                    rel={direction}
                >
                    {
                        direction === SIDE_ITEM_DIRECTION.PREV ?
                        <React.Fragment>
                            {renderArrow()}
                            {renderTitle()}
                        </React.Fragment> :
                        <React.Fragment>
                            {renderTitle()}
                            {renderArrow()}
                        </React.Fragment>
                    }
                </Link>
            </li>
        </ul>
    );
};

SideLinkItem.propTypes = {
    direction: PropTypes.oneOf([
        SIDE_ITEM_DIRECTION.PREV,
        SIDE_ITEM_DIRECTION.NEXT,
    ]).isRequired,
    to: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};

export default SideLinkItem;
