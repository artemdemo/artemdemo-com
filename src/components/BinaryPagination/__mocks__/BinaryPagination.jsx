import React from 'react';

const BinaryPagination = props => (
    <div data-mock='BinaryPagination'>
        {JSON.stringify(props, null, 2)}
    </div>
);

export default BinaryPagination;