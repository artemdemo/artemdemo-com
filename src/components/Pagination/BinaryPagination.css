@media (max-width: 768px) {
    .binary-pagination {
        display: block;
    }
}

.binary-pagination-link {
    max-width: 200px;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
}

@media (min-width: 992px) {
    .binary-pagination-link {
        max-width: 300px;
    }
}