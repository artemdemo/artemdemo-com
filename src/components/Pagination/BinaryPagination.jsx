import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';

class BinaryPagination extends React.PureComponent {
    renderPrev() {
        const { previous } = this.props;
        if (previous && previous.slug != null) {
            const title = previous.title || previous.slug;
            return (
                <li>
                    <Link to={previous.slug} rel="prev">
                        ← {title}
                    </Link>
                </li>
            );
        }
        return null;
    }

    renderNext() {
        const { next } = this.props;
        if (next && next.slug != null) {
            const title = next.title || next.slug;
            return (
                <li>
                    <Link to={next.slug} rel="next">
                        {title} →
                    </Link>
                </li>
            );
        }
        return null;
    }

    render() {
        return (
            <ul
                style={{
                    display: 'flex',
                    flexWrap: 'wrap',
                    justifyContent: 'space-between',
                    listStyle: 'none',
                    padding: 0,
                }}
            >
                {this.renderPrev()}
                {this.renderNext()}
            </ul>
        );
    }
}

BinaryPagination.propTypes = {
    previous: PropTypes.shape({
        slug: PropTypes.string,
        title: PropTypes.string,
    }),
    next: PropTypes.shape({
        slug: PropTypes.string,
        title: PropTypes.string,
    }),
};

BinaryPagination.defaultProps = {
    previous: null,
    next: null,
};

export default BinaryPagination;