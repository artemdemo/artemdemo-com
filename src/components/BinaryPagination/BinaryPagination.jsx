import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import SideLinkItem, { SIDE_ITEM_DIRECTION } from './SideLinkItem';

import './BinaryPagination.css';

const PaginationContainer = styled.div`
    &::after {
        content: '';
        display: table;
        clear: both;
    }
    
    @media (max-width: 768px) {
        text-align: center;
    }
`;

const PaginationItem = styled.div`
    ${props => props.left && `
        float: left;    
    `};
    ${props => props.right && `
        float: right;
    `};
    @media (max-width: 768px) {
        float: initial;
        display: inline-block;
        width: 51%;
    }
`;

class BinaryPagination extends React.PureComponent {
    renderPrev() {
        const { previous } = this.props;
        if (previous && previous.slug != null) {
            const title = previous.title || previous.slug;
            return (
                <SideLinkItem
                    direction={SIDE_ITEM_DIRECTION.PREV}
                    to={previous.slug}
                    title={title}
                />
            );
        }
        return null;
    }

    renderNext() {
        const { next } = this.props;
        if (next && next.slug != null) {
            const title = next.title || next.slug;
            return (
                <SideLinkItem
                    direction={SIDE_ITEM_DIRECTION.NEXT}
                    to={next.slug}
                    title={title}
                />
            );
        }
        return null;
    }

    render() {
        return (
            <PaginationContainer>
                <PaginationItem left>
                    {this.renderPrev()}
                </PaginationItem>
                <PaginationItem right>
                    {this.renderNext()}
                </PaginationItem>
            </PaginationContainer>
        );
    }
}

BinaryPagination.propTypes = {
    previous: PropTypes.shape({
        slug: PropTypes.string,
        title: PropTypes.string,
    }),
    next: PropTypes.shape({
        slug: PropTypes.string,
        title: PropTypes.string,
    }),
};

BinaryPagination.defaultProps = {
    previous: null,
    next: null,
};

export default BinaryPagination;
