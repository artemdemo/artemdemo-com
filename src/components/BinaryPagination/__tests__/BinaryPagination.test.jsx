import React from 'react';
import renderer from 'react-test-renderer';
import BinaryPagination from '../BinaryPagination';

jest.mock('gatsby');
jest.mock('../SideLinkItem');

describe('BinaryPagination', () => {
    it('should render empty', () => {
        const tree = renderer.create(
            <BinaryPagination />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('shouldn\'t render without slug', () => {
        const tree = renderer.create(
            <BinaryPagination
                previous={{
                    title: 'Prev title',
                }}
                next={{
                    title: 'Next title',
                }}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should use slug if title not provided', () => {
        const tree = renderer.create(
            <BinaryPagination
                previous={{
                    slug: '/prev',
                }}
                next={{
                    slug: '/next',
                }}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('should render with data', () => {
        const tree = renderer.create(
            <BinaryPagination
                previous={{
                    slug: '/prev',
                    title: 'Prev title',
                }}
                next={{
                    slug: '/next',
                    title: 'Next title',
                }}
            />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });
});