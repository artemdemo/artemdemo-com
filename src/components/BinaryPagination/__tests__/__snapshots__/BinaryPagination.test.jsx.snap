// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BinaryPagination should render empty 1`] = `
<div
  className="BinaryPagination__PaginationContainer-sc-35l4ss-0 bnEHjt"
>
  <div
    className="BinaryPagination__PaginationItem-sc-35l4ss-1 fjGTlq"
  />
  <div
    className="BinaryPagination__PaginationItem-sc-35l4ss-1 ibbqOI"
  />
</div>
`;

exports[`BinaryPagination should render with data 1`] = `
<div
  className="BinaryPagination__PaginationContainer-sc-35l4ss-0 bnEHjt"
>
  <div
    className="BinaryPagination__PaginationItem-sc-35l4ss-1 fjGTlq"
  >
    <div
      data-mock="SideLinkItem"
    >
      {
  "direction": "prev",
  "to": "/prev",
  "title": "Prev title"
}
    </div>
  </div>
  <div
    className="BinaryPagination__PaginationItem-sc-35l4ss-1 ibbqOI"
  >
    <div
      data-mock="SideLinkItem"
    >
      {
  "direction": "next",
  "to": "/next",
  "title": "Next title"
}
    </div>
  </div>
</div>
`;

exports[`BinaryPagination should use slug if title not provided 1`] = `
<div
  className="BinaryPagination__PaginationContainer-sc-35l4ss-0 bnEHjt"
>
  <div
    className="BinaryPagination__PaginationItem-sc-35l4ss-1 fjGTlq"
  >
    <div
      data-mock="SideLinkItem"
    >
      {
  "direction": "prev",
  "to": "/prev",
  "title": "/prev"
}
    </div>
  </div>
  <div
    className="BinaryPagination__PaginationItem-sc-35l4ss-1 ibbqOI"
  >
    <div
      data-mock="SideLinkItem"
    >
      {
  "direction": "next",
  "to": "/next",
  "title": "/next"
}
    </div>
  </div>
</div>
`;

exports[`BinaryPagination shouldn't render without slug 1`] = `
<div
  className="BinaryPagination__PaginationContainer-sc-35l4ss-0 bnEHjt"
>
  <div
    className="BinaryPagination__PaginationItem-sc-35l4ss-1 fjGTlq"
  />
  <div
    className="BinaryPagination__PaginationItem-sc-35l4ss-1 ibbqOI"
  />
</div>
`;
